### ISSUE TITLE
Make the date+time in the JSON report for "validatedAt" a standard format

### ISSUE THREAD

**Issue by bradyhunsaker** [MobilityData/gtfs-validator#1652]
### Describe the problem

The date+time for when validation is run uses a custom format, like "2024-01-15 at 23:14:46 EST". This is not a standard format and so is not easily machine readable.

It is put in two places: the JSON report as "validatedAt" and the HTML report in the first sentence.

### Proposed solution

For the JSON report, it would be better to use a standard format, like ISO_ZONED_DATE_TIME: 2024-01-15T24:14:45+05:00[America/New_York]

### Alternatives you've considered

It could be left as is. That makes it harder for anyone parsing the JSON report but doesn't break any existing parsers.

### Additional context

This was noticed during review of issue #1292.

The main concern is that this would break any existing parsers of the validatedAt field in the JSON report.
My guess is that there are few if any such parsers and that a one-time break to use a standard format would be worthwhile.

The custom format is specified in the code here:
https://github.com/MobilityData/gtfs-validator/blob/86f9484480a2a8f64d035be2a144f809df80051b/main/src/main/java/org/mobilitydata/gtfsvalidator/runner/ValidationRunner.java#L270

**Comment by davidgamez** [comment]
Tasks:
- Investigate the proper Java classes to use.
- Change the date format to ISO 8601 on JSON and HTML report, Example: _`2024-01-15T24:14:45+05:00`_. Timestamp from JVM's timezone or parameter.
- Review if there is any other code that uses Datetime format.

**Comment by qcdyx** [comment]
- I searched for java.util.Calendar and java.util.Date (the old java Data/Time class) - No results found. 
- I used DateTimeFormatter.ISO_DATE_TIME as the formatter. This will format the date and time as a string in the format yyyy-MM-dd'T'HH:mm:ss.SSSZ, which is the standard ISO 8601 format. For example, "2022-03-14T09:30:00.000Z".
![image](https://github.com/MobilityData/gtfs-validator/assets/5789435/c302ac0a-bead-4238-9f86-88df69c7ca90)
